.github/workflows/ci.yaml

```diff
@@ -0,0 +1,52 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+# The goal here is to cancel older workflows when a PR is updated (because it's pointless work)
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  unittest:
+    name: unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: pip
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest pytest-cov
+
+      - name: Test with pytest
+        run: |
+          python -m pytest --cov=fhirclient --cov-report=xml
+
+      - name: Log missing coverage
+        run: |
+          coverage report -m --skip-covered
+
+      - name: Check coverage report
+        if: github.ref != 'refs/heads/main'
+        uses: orgoro/coverage@v3.2
+        with:
+          coverageFile: coverage.xml
+          token: ${{ secrets.GITHUB_TOKEN }}
+          thresholdAll: .35
+          thresholdNew: 1
+          thresholdModified: 0
```

Comment:
So hear me out. I'm excluding the generated models/* folder - which... I could believe I shouldn't. But I'm taking baby steps. The generated code does miss some stuff - import error handling and some small enum-style classes that don't get instantiated during the tests.

So I'm not super stressed about that coverage gap, and I'd rather come back to that later and focus on the coverage of the manual code we wrote for this repo. Which... isn't great:

```
File 	statements 	missing 	excluded 	coverage
fhirclient/__init__.py 	5 	0 	0 	100%
fhirclient/auth.py 	249 	138 	0 	45%
fhirclient/client.py 	138 	138 	0 	0%
fhirclient/server.py 	150 	74 	0 	51%
Total                  	542 	350 	0 	35%
```

So with this coverage config, I'm just surfacing the problem, and making sure it doesn't get much worse - but not really enforcing a ratchet upwards.

Reply:
oh, i'm totally here for 'start with a bar you can easily clear and work back from there'